### PR TITLE
chore: Updating Azure NPM to v1.0.31

### PR DESCRIPTION
**Pull request by jaer-tsun**
**Reason for Change**:

Updates to the latest release of Azure NPM:
https://github.com/Azure/azure-container-networking/releases/tag/v1.0.31

**Issue Fixed**:

**Requirements**:

- [ ] uses [conventional commit messages](https://www.conventionalcommits.org/)
  
- [ ] includes documentation
- [ ] adds unit tests
- [ ] tested upgrade from previous version

**Notes**:

### CODE REVIEW

**Comment by jaer-tsun** [comment]
@jackfrancis 
@mboersma

**Comment by jaer-tsun** [comment]
oops, let me push the npm image first...

**Comment by jaer-tsun** [comment]
ok I pushed NPM v1.0.31

File: packer/install-dependencies.sh [hunk 1]
```diff
@@ -313,9 +313,9 @@ for AZURE_CNI_NETWORKMONITOR_VERSION in ${AZURE_CNI_NETWORKMONITOR_VERSIONS}; do
 done
 
 AZURE_NPM_VERSIONS="
+1.0.31
```
Comment (CecileRobertMichon) [inline packer/install-dependencies.sh]: this is only adding the new version to the VHD build, not using it in code... I assume you also intended to use this new npm version by default?

Comment (jaer-tsun) [reply packer/install-dependencies.sh]: yup I would like to use it by default

Comment (CecileRobertMichon) [reply packer/install-dependencies.sh]: you also need to change https://github.com/Azure/aks-engine/blob/master/pkg/api/k8s_versions.go#L46 then

Comment (jaer-tsun) [reply packer/install-dependencies.sh]: let me check the previous pr again

**Comment by mboersma** [comment]
@jaer-tsun I mirrored the azure-vnet-cni 1.0.31 binaries to our production blob store. ✔️

File: packer/install-dependencies.sh [hunk 2]
```diff
@@ -313,9 +313,9 @@ for AZURE_CNI_NETWORKMONITOR_VERSION in ${AZURE_CNI_NETWORKMONITOR_VERSIONS}; do
...
 1.0.30
 1.0.29
-1.0.28
```
Comment (jackfrancis) [inline packer/install-dependencies.sh]: Let's not remove 1.0.28 from the VHD as part of this change

Comment (jaer-tsun) [reply packer/install-dependencies.sh]: Is it ok to remove since 1.0.28 and below had quite a few bugs?

File: packer/install-dependencies.sh [hunk 3]
```diff
@@ -313,9 +313,10 @@ for AZURE_CNI_NETWORKMONITOR_VERSION in ${AZURE_CNI_NETWORKMONITOR_VERSIONS}; do
...
+v.0.28
```
Comment (jackfrancis) [inline packer/install-dependencies.sh]: typo

**Pull request merged**